### PR TITLE
feat(config): ngConfig

### PR DESCRIPTION
**Pull request by jkuri**

### CODE REVIEW

File: addon/ng2/commands/set.ts [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
     { name: 'global', type: Boolean, default: false, aliases: ['g'] },
   ],
 
   run: function (commandOptions, rawArgs): Promise<void> {
     return new Promise(resolve => {
-      const config = new CliConfig();
-      config.set(rawArgs[0], rawArgs[1], commandOptions.force);
+      let config = new Config();
+      config.validatePath(rawArgs[0]);
+      config.set(config.config, rawArgs[0], rawArgs[1]);
```
Comment (hansl) [inline addon/ng2/commands/set.ts]: Could you do a `a=b` kinda of syntax?

Basically:
```typescript
for (let arg of rawArgs) {
  let [key, value] = arg.split('=', 1);
  config.validatePath(key);
  config.set(key, value);
}```

What do you think?  This way we can `ng config set key[1]=value1 key[2]=value2`.

File: tests/models/config.spec.ts [hunk 2]
```diff
@@ -0,0 +1,449 @@
+import {Config} from '../../addon/ng2/models/config';
+
+const fs      = require('fs');
+const path    = require('path');
+const expect  = require('chai').expect;
+const schema  = path.resolve(process.cwd(), 'test.json');
```
Comment (hansl) [inline tests/models/config.spec.ts]: Missing file?

**Comment by hansl** [comment]
Linting and some tests are failing.

File: addon/ng2/commands/set.ts (hunk 1 above)
Comment (jkuri) [inline addon/ng2/commands/set.ts]: Good point, thanks. Done.

File: tests/models/config.spec.ts (hunk 2 above)
Comment (jkuri) [inline tests/models/config.spec.ts]: Just a path to the file that is used for testing purposes, it is created `beforeEach`.

Comment (hansl) [inline tests/models/config.spec.ts]: yes, but where is that file in this pr?

Comment (jkuri) [inline tests/models/config.spec.ts]: It's not there as it is not needed it is [created](https://github.com/jkuri/angular-cli/blob/ng-config/tests/models/config.spec.ts#L18) before each test is run and removed [after](https://github.com/jkuri/angular-cli/blob/ng-config/tests/models/config.spec.ts#L24) test is done.

Comment (hansl) [inline tests/models/config.spec.ts]: Oh so you're not testing any actual schema, just an empty one?

**Comment by Brocco** [comment]
@jkuri You'll also want to pull in the changes I have in this PR (#468) as part of this PR. It will expose the configuration on the project object for access within our commands and blueprints.

**Pull request closed without merging**